src/utils/marketxyz/index.ts

```diff
@@ -78,15 +78,36 @@ export const fetchGasForCall = async (
   };
 };
 
-const delay = (t: number) => {
-  return new Promise((resolve) => {
-    setTimeout(resolve, t);
-  });
+export const checkCTokenisApproved = async (
+  asset: USDPricedPoolAsset,
+  amount: number,
+  address: string,
+) => {
+  const cToken = asset.cToken;
+  const sdk = cToken.sdk;
+  const amountBN = sdk.web3.utils.toBN(
+    amount * 10 ** asset.underlyingDecimals.toNumber(),
```

Comment:
Please scan the whole code base for multiplication like this I have a feeling this will cause overflow